### PR TITLE
Dynamo Var_mean dynamo pass

### PR DESCRIPTION
**Pull request by apbose**
Var_mean lowering pass

### CODE REVIEW

File: py/torch_tensorrt/dynamo/lowering/_decompositions.py [hunk 1]
```diff
@@ -76,12 +126,49 @@ def addmm_replacement(
...
+)
 def reciprocal_replacement(
     input_: torch.Tensor,
 ) -> torch.Tensor:
     return torch.div(1, input_)
 
 
-def get_decompositions() -> Dict[OpOverload, Callable[..., Any]]:
-    return DECOMPOSITIONS
+@register_decomposition(torch.ops.aten.var_mean, registry=TORCH_TRT_DECOMPOSITIONS)
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/lowering/_decompositions.py:138]: Remove this registration, since it causes duplicate registration errors because `var_mean` with no overload encapsulates _all_ `var_mean` overloads.

File: py/torch_tensorrt/dynamo/lowering/_decompositions.py [hunk 2]
```diff
@@ -76,12 +126,49 @@ def addmm_replacement(
...
+@register_decomposition(torch.ops.aten.var_mean.dim, registry=TORCH_TRT_DECOMPOSITIONS)
+@register_decomposition(torch.ops.aten.var_mean.correction, registry=TORCH_TRT_DECOMPOSITIONS)
+def var_mean_replacement(
+    input_: torch.Tensor,
+    dim: int = None,
+    correction: int = 1,
+    keepdim: bool = False,
+    unbiased: bool = True,
+) -> List[torch.Tensor]:
+    if dim is not None:
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/lowering/_decompositions.py:148]: I think this should be `if dim is None`

File: py/torch_tensorrt/dynamo/lowering/_decompositions.py [hunk 3]
```diff
@@ -76,12 +126,49 @@ def addmm_replacement(
...
+        mean = torch.mean(input_)
+        variance = torch.mean(torch.pow(torch.sub(input_, mean), 2))
+    else:
+        mean = torch.mean(input_, dim)
+        variance = torch.mean(torch.pow(torch.sub(input_, mean), 2), dim)
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/lowering/_decompositions.py:153]: Both of these `torch.mean` calls need a `keepdim` as their final argument

File: py/torch_tensorrt/dynamo/lowering/_decompositions.py [hunk 4]
```diff
@@ -76,12 +126,49 @@ def addmm_replacement(
...
 
 
-def get_decompositions() -> Dict[OpOverload, Callable[..., Any]]:
-    return DECOMPOSITIONS
+@register_decomposition(torch.ops.aten.var_mean, registry=TORCH_TRT_DECOMPOSITIONS)
+@register_decomposition(torch.ops.aten.var_mean.dim, registry=TORCH_TRT_DECOMPOSITIONS)
+@register_decomposition(torch.ops.aten.var_mean.correction, registry=TORCH_TRT_DECOMPOSITIONS)
+def var_mean_replacement(
+    input_: torch.Tensor,
+    dim: int = None,
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/lowering/_decompositions.py:143]: Switch to `Optional[int]` to allow `None`
